texar/core/attention_mechanism.py

```diff
@@ -330,24 +275,21 @@ def __init__(self,
                                                    torch.Tensor]] = None,
                  score_mask_value: Optional[torch.Tensor] = None):
         # For LuongAttention, we only transform the memory layer; thus
-        # num_units **must** match expected the query depth.
+        # num_units must match expected the query depth.
         if probability_fn is None:
             probability_fn = lambda x: F.softmax(x, dim=-1)
         self.wrapped_probability_fn = lambda score, _: probability_fn(score)
+
         super(LuongAttention, self).__init__(
-            query_layer=None,
             memory_layer=nn.Linear(encoder_output_size, num_units, False),
+            query_layer=None,
             score_mask_value=score_mask_value)
-        self._num_units = num_units
         self._encoder_output_size = encoder_output_size
```

Comment:
Since `encoder_output_size` is set in all subclasses, it might be better to make it an argument of the base class constructor, so we could remove `Optional` from the type.